components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -1423,4 +1427,83 @@ TEST_F(SolanaTxManagerUnitTest, ProcessSolanaHardwareSignature) {
                                      "");
 }
 
+TEST_F(SolanaTxManagerUnitTest, RebroadcastTransaction) {
+  auto data = GetSystemProgramTransferTxData(sol_account(), to_account());
+  ASSERT_TRUE(data);
+  SetInterceptor(latest_blockhash1_, last_valid_block_height1_, "", "", true,
+                 last_valid_block_height1_, "[null]");
+
+  std::string meta_id1;
+  AddUnapprovedTransaction(mojom::kSolanaMainnet, data.Clone(), sol_account(),
+                           &meta_id1);
+  ASSERT_FALSE(meta_id1.empty());
+
+  // No rebroadcast when first sending out the tx.
+  ApproveTransaction(meta_id1);
+  task_environment_.RunUntilIdle();
```

Comment:
`ApproveTransaction` already has run loop to wait for the callback, what are we waiting for here exactly?

Reply:
This is waiting for tx to be updated, ApproveTransaction callback is called without waiting on the result of UpdatePendingTransactions triggered in SolanaTxManager::OnSendSolanaTransaction.

Reply:
Can we do
```
void WaitForUpdatePendingTransactions() {
  task_environment_.RunUntilIdle();
}
```
so we can easily tell what it is waiting for without comments and 
if we have a better way to wait for that, it can be easy to be spotted and changed.

Reply:
Fixed in https://github.com/brave/brave-core/pull/23329/commits/7cc18ad5a5f0ef1ca890d6eb24f5d3da6a50a255